app/src/main/java/com/edu/uni/augsburg/uniatron/stepcounter/BroadcastReceiverOnStateChanged.java

```diff
@@ -0,0 +1,36 @@
+package com.edu.uni.augsburg.uniatron.stepcounter;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+
+/**
+ * The BroadCastReceiver starts the StepCountService as soon as boot is completed.
+ *
+ * @author Leon Wöhrl
+ */
+public class BroadcastReceiverOnStateChanged extends BroadcastReceiver {
+    @Override
+    public void onReceive(Context context, Intent intent) {
+
+        String action = intent.getAction();
+        // on device restart, app install, app update
+        if (action.equalsIgnoreCase(Intent.ACTION_BOOT_COMPLETED)
+                || action.equalsIgnoreCase(Intent.ACTION_PACKAGE_REPLACED)
+                || action.equalsIgnoreCase(Intent.ACTION_PACKAGE_ADDED)) {
+
+                startService(context);
+        }
+    }
+
+    private void startService(Context context) {
+        // fixes crash on post Android O devices; services cannot be started in background!
```

Comment:
See the [official documentation](https://developer.android.com/about/versions/oreo/background)